### PR TITLE
Jenkins: Polling plugin for Ontrack

### PR DESCRIPTION
**Pull request by cpf**
See commits :) Plx give some feedback

### CODE REVIEW

File: ontrack-jenkins/src/main/java/org/jenkinsci/plugins/ontrack/OntrackValidationStampPollingTrigger.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+												String validationStamp) throws ANTLRException {
+		super(cronTabSpec, triggerLabel);
+		this.project = project;
+		this.branch = branch;
+		this.validationStamp = validationStamp;
+	}
+
+	@Override
+	protected File getLogFile() {
+		return new File(job.getRootDir(), "ontrack-validationstamp-polling-trigger.log");
```
Comment (dcoraboeuf) [inline ontrack-jenkins/src/main/java/org/jenkinsci/plugins/ontrack/OntrackValidationStampPollingTrigger.java:36]: If you use `File`, your plug-in won't be able to run on a slave, only onto the master. What you can do is use the `AbstractBuild.getWorkspace()` or similar, and the associated `FilePath` API, which enables IO in a distributed environment.

File: ontrack-jenkins/src/main/java/org/jenkinsci/plugins/ontrack/OntrackValidationStampPollingTrigger.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+				logException(xTriggerLog, e);
+			}
+		}
+
+		return false;
+	}
+
+	@Override
+	protected void start(Node pollingNode, BuildableItem project, boolean newInstance, XTriggerLog xTriggerLog) throws XTriggerException {
+		this.lastBuildNrFile = new File(project.getRootDir().getAbsolutePath() + "/lastBuildNr");
```
Comment (dcoraboeuf) [inline ontrack-jenkins/src/main/java/org/jenkinsci/plugins/ontrack/OntrackValidationStampPollingTrigger.java]: Same remark than before.

File: ontrack-jenkins/src/main/java/org/jenkinsci/plugins/ontrack/OntrackValidationStampPollingTrigger.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	public void setLastBuildNr(String lastBuildNr) {
+		this.lastBuildNr = lastBuildNr;
+	}
+
+	private void saveLastBuildNr(String lastBuildNr, XTriggerLog xTriggerLog) throws IOException {
+		if (!lastBuildNrFile.exists()) {
+			lastBuildNrFile.createNewFile();
+		}
+
+		FileWriter fw = new FileWriter(lastBuildNrFile);
```
Comment (dcoraboeuf) [inline ontrack-jenkins/src/main/java/org/jenkinsci/plugins/ontrack/OntrackValidationStampPollingTrigger.java]: To be tested on a slave.

File: ontrack-jenkins/src/main/java/org/jenkinsci/plugins/ontrack/OntrackValidationStampPollingTrigger.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+		BufferedWriter bw = new BufferedWriter(fw);
+		bw.write(lastBuildNr);
+		bw.close();
```
Comment (dcoraboeuf) [inline ontrack-jenkins/src/main/java/org/jenkinsci/plugins/ontrack/OntrackValidationStampPollingTrigger.java]: Use `try / finally` to close the input streams.

File: ontrack-jenkins/src/main/java/org/jenkinsci/plugins/ontrack/OntrackValidationStampPollingTrigger.java [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+		fw.close();
+		xTriggerLog.info(String.format("Wrote buildNr: %s", lastBuildNr));
+	}
+
+	private void loadLastBuildNr(XTriggerLog xTriggerLog) throws IOException {
+		FileReader fr = new FileReader(lastBuildNrFile);
```
Comment (dcoraboeuf) [inline ontrack-jenkins/src/main/java/org/jenkinsci/plugins/ontrack/OntrackValidationStampPollingTrigger.java]: To be tested on a slave.

File: ontrack-jenkins/src/main/java/org/jenkinsci/plugins/ontrack/OntrackValidationStampPollingTrigger.java [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+		BufferedReader br = new BufferedReader(fr);
+
+		lastBuildNr = br.readLine();
+
+		br.close();
```
Comment (dcoraboeuf) [inline ontrack-jenkins/src/main/java/org/jenkinsci/plugins/ontrack/OntrackValidationStampPollingTrigger.java]: Use `try / finally` to close the input streams.

**Comment by cpf** [comment]
Actually, I was just starting that xD I'll still test those things...

**Comment by dcoraboeuf** [comment]
No problem, I can always pull the new version :)

**Pull request merged**